File: Daily-Temperatures.java
1class Solution {
2    public int[] dailyTemperatures(int[] temperatures) 
3    {
4        Deque<int[]> stack = new ArrayDeque<int[]>();
5        int[] ans = new int[temperatures.length];
6        Arrays.fill(ans, 0);
7        stack.push(new int[]{temperatures[0], 0});
8        for(int i= 1;i<temperatures.length;i++)
9        {
10            while(!stack.isEmpty() && stack.peek()[0] < temperatures[i])
11            {
12                int[] ele =  stack.peek();
13                //System.out.println(ele[0] + " "+ ele[1]);
14                ans[ele[1]] = i - ele[1];
15                stack.pop();
16            }
17            stack.push(new int[]{temperatures[i], i});
18        }       
19        return ans;
20
21    }
22}

File: Daily-Temperatures.cpp
1class Solution {
2public:
3    vector<int> dailyTemperatures(vector<int>& temperatures) 
4    {
5        stack<pair<int,int>> tempStack;
6        vector<int> ans;
7        for(int i=0;i<temperatures.size();i++)
8        {
9            ans.push_back(-1);
10        }
11        for(int i=0;i<temperatures.size();i++)
12        {
13            while(!tempStack.empty() &&tempStack.top().first < temperatures[i] )
14            {
15                pair<int, int> node = tempStack.top();
16                ans[node.second] = i - node.second;
17                tempStack.pop();
18            }
19
20            tempStack.push({temperatures[i], i});
21        }
22        for(int i=0;i<ans.size();i++)
23        {
24            if(ans[i] == -1)
25                ans[i] =0;
26        }
27        return ans;
28    }
29};